sycl/doc/design/OffloadDesign.md

```diff
@@ -296,6 +296,125 @@ resemble `--gpu-tool-arg=<arch> <arg>`.  This corresponds to the existing
 option syntax of `-fsycl-targets=intel_gpu_arch` where `arch` can be a fixed
 set of targets.
 
+#### --offload-arch
```

Comment:
I think this section is describing the command line arguments to the `clang-linker-wrapper` tool, not the command line options to the compiler driver.  Correct?

If that is the case, then I think the wording in the section above needs to be updated because that section implies that `--gpu-tool-arg` is used to specify the device name.  One of the examples shows:

> `--gpu-tool-arg="-device skl -options -extraopt_skl"`

But, I guess the intent now is to pass these flags instead:

> --offload-arch=skl --gpu-tool-arg="-options -extraopt_skl"

Is that right?

BTW, when will we update the documentation in UsersManual.md to describe `--offload-arch`?  I suppose we want to wait until the new offload design is fully implemented?

Reply:
> If that is the case, then I think the wording in the section above needs to be updated because that section implies that --gpu-tool-arg is used to specify the device name. One of the examples shows:
> 
> --gpu-tool-arg="-device skl -options -extraopt_skl"
> 
> But, I guess the intent now is to pass these flags instead:
> 
> --offload-arch=skl --gpu-tool-arg="-options -extraopt_skl"
> 
> Is that right?

AFAIC, `--gpu-tool-arg` only captures the "options" passed to `-Xsycl-target-backend`  and is later passed on to  the `clang-linker-wrapper` tool.

Example:

```
clang++  -fsycl --offload-new-driver  -fsycl-targets=spir64_gen  -Xsycl-target-backend=spir64_gen -backend-gen-opt  syclfile.cpp
clang-linker-wrapper     "--gpu-tool-arg=-backend-gen-opt"
```


```
clang++ -### -fsycl -fsycl-targets=intel_gpu_skl  -Xsycl-target-backend=intel_gpu_skl "-options -extraopt_skl"  --offload-new-driver syclfile.cpp
"clang-linker-wrapper  --gpu-tool-arg=-options -extraopt_skl"
```

Based on the above example,` --gpu-tool-arg` does not specify the device name. It appears to just capture the options passed to `Xsycl-target-backend`

And in the below example, I'm not sure `--gpu-tool-arg="-device skl"` is correct.
Maybe @mdtoguchi  can clarify?
> -fsycl -fsycl-targets=spir64_gen,intel_gpu_skl
-Xsycl-target-backend=spir64_gen "-device pvc -options -extraopt_pvc"
-Xsycl-target-backend=intel_gpu_skl "-options -extraopt_skl"

*Example: spir64_gen enabling options*

> --gpu-tool-arg="-device pvc -options extraopt_pvc"
--gpu-tool-arg="-device skl -options -extraopt_skl"

*Example: clang-linker-wrapper options*

My section on `--offload-arch` just describes the driver option that users could use instead of `-fsycl-targets` option.

Example:

```
$ clang++ -fsycl -offload-arch=bdw --offload-new-driver foo.cpp -c // SYCL AOT for Intel GPU.
$ clang++ -fsycl -offload-arch=broadwell --offload-new-driver foo.cpp -c // SYCL AOT for Intel CPU.
```

> I suppose we want to wait until the new offload design is fully implemented?

That is my understanding, as the `offload-arch` implementation is still not available.

Reply:
The expectation is for these values to be added to the `clang-offload-packager` information as part of the fat object.  The `arch` information along with the associated `spir64_gen` triple value will inform the `clang-linker-wrapper` to use the `arch` value with the corresponding `ocloc -device <arch>` call.

Reply:
@mdtoguchi 
I was referring to the example that was already in the document comparing `-Xsycl-target-backend` options and the corresponding `--gpu-tool-arg` value.
Particularly, this one:

> -Xsycl-target-backend=intel_gpu_skl "-options -extraopt_skl"
> --gpu-tool-arg="-device skl -options -extraopt_skl"

But when I checked the invocation, I did not see `-device skl` captured by `--gpu-tool-arg`
So I was wondering if that was a typo/mistake.

Reply:
The behavior witnessed (not passing `-device skl` to `--gpu-tool-arg`) is correct.  I was referring to how the arch information is actually passed for consumption.

Reply:
For `-Xsycl-target-backend=intel_gpu_skl "-options -extraopt_skl"` , we have documented `--gpu-tool-arg="-device skl -options -extraopt_skl" ` Lines 280-288 in the original document.

But I do not see `-device skl `captured in `--gpu-tool-arg`.
Should we update the document and remove this?

Reply:
It makes sense to update the document in that aspect.  Thanks!